substrate/frame/revive/rpc/src/receipt_provider.rs

```diff
@@ -305,7 +432,14 @@ impl<B: BlockInfoProvider> ReceiptProvider<B> {
 				qb.push(" AND block_number <= ").push_bind(to_block.as_u64() as i64);
 			},
 			(None, None, Some(hash)) => {
-				qb.push(" AND block_hash = ").push_bind(hash.0.to_vec());
+				// Try to resolve Ethereum hash to Substrate hash first
+				let substrate_hash = if let Some(resolved) = self.get_substrate_hash(&hash).await {
+					resolved
+				} else {
+					// Fallback: treat as Substrate hash for backward compatibility
+					hash
+				};
+				qb.push(" AND block_hash = ").push_bind(substrate_hash.0.to_vec());
```

Comment:
why do we need the substrate_hash here? isn't the user quering with the eth blockhash?

Reply:
Logs are inserted with Substrate block hash. User should query with Ethereum block hash, therefore we try to resolve it to the Substrate block hash.
If not possible we use provided hash as a Substrate block hash -  as a fallback.